### PR TITLE
Add instructions on Travis link checking to editorial workflow

### PR DESCRIPTION
**Pull request by mdlincoln**
Now that we have Travis CI running constant checks of internal and external links, it will begin catching typos on newly published lessons as soon as editors commit them to the main repository. I think it would be a help for editors to check for build errors as part of their final publishing workflow, since it has happened that URL typos in new lessons and in translations have led to some broken links making their way online.

I hope this doesn't read as an onerous addition. I added some screenshots, since navigating the Travis CI build results can be intimidating your first time around. In most cases, the tests ought to pass fine, and editors will not even need to pull up the Travis site.

### CODE REVIEW

**Comment by mdlincoln** [comment]
FYI @jerielizabeth I just tagged you since I knew you have been working on editor workflow management - but I'm glad for comments from anyone!

**Comment by acrymble** [comment]
this should be done on the submissions repo, no? Before the files get moved?

Will this also catch missing elements from the YAML?

The editorial checklist is getting rather long. I wonder if we don't need someone to act as a typesetter as distinct from editor? I imagine it would iron out some inconsistencies and reduce the burden on editors.

**Comment by mdlincoln** [comment]
Good questions.

1. We could add Travis CI checking to the submissions repo as well. However, #398 needs to be finished (cc @wcaleb), so that old submissions are not constantly re-checked.
2. It will only catch formal errors in the YAML (aka someone put in an illegal quote mark), not semantic errors like leaving out a field. We could probably add an additional routine to the Travis script to check for required YAML fields, but I'd want to scope out how much work that would involve.

**Comment by wcaleb** [comment]
I need to get on #398, thanks for the reminder. I would be willing to serve in the "typesetter" kind of role if we end up going in that direction. At this point scrubbing Markdown, fixing errors and the like is one of the main things I do already.

**Comment by mdlincoln** [comment]
Ditto on scrubbing/tidying the text and code in this repo - I'm already doing a lot of that as well and am happy to continue.

I do think it's important for editors to be aware of these tests at the time of publishing, because it's often easier and more appropriate for editors to fix bad links at that stage in the process. They will be familiar with the lesson text and the intent/context of a link, and have closer contact with the author, and thus be in a better position to repair mistakes intelligently.

Thus, I think a "typesetter" should, in this case, be reactive, helping to catch any mistakes the editor didn't/can't fix, rather than having to assume responsibility for _all lesson metadata and formatting_. I feel like the latter would just devolve into me pinging an editor with a dozen queries for things that the editor could have just filled out in the first place - duplicating effort, rather than reducing it.

File: editor-guidelines.md [hunk 1]
```diff
@@ -198,6 +198,47 @@ Upload the original image to the [gallery/originals](https://github.com/programm
...
+
+We run these builds primarily to check that URLs that _once_ were functional are _still_ functional, as often times external web pages are moved to new addresses, or are no longer live.
+They are also an excellent way to catch small typos that may have evaded authors, editors, and reviewers.
+The status of these tests (often called a "Build Status" on Travis CI and on GitHub) can be seen by navigating to the [programminghistorian repository page][ph_repo], and clicking "Commits" on the upper left side of the code menu.
+
+![GitHub commit menu location](images/editor-guidelines/gh_commits_location_screen.png)
+
+This will show you the list of every change made to the main repository, along with a status icon:
+
+- Green check mark: you're good to go! All the links on the page were checked and found valid. **You may skip the rest of this section.**
```
Comment (jerielizabeth) [inline editor-guidelines.md]: Quick thought: what about an internal link to the end of the section inside the "You may skip the rest of this section"?

**Review by jerielizabeth — COMMENTED** [review]
I think it looks great! Just the one suggestion for ease of navigation for editors with error-free commits.

**Comment by jerielizabeth** [comment]
If it's not too much work, I think having Travis check the required YAML fields would be useful. There are quite a few details to keep track of and I think the automated check would be the most reliable way to verify that we are all using the same fields (as opposed to another human reader.)

File: editor-guidelines.md (hunk 1 above)
Comment (mdlincoln) [reply editor-guidelines.md]: good call! Will add.

**Comment by mdlincoln** [comment]
Alright, I'll look in to developing some YAML check to add in to this script. I wouldn't really implement and deploy it, though, until #414 is finished and merged, since there may be some YAML changes because of that. I'll open a new to-do issue for that, but hold off work until the design changes are merged.

**Comment by mdlincoln** [comment]
Flagging spanish language team for translation @arojascastro, @mariajoafana,
@vgayolrs

**Comment by arojascastro** [comment]
Hello @mdlincoln  @jerielizabeth I am sorry, but I cannot figure out what changes need to be done in our guidelines for editors. I cannot find your changes here http://programminghistorian.org/editor-guidelines  

I know that I can see the changes you made in this pull request, but I would like to check the whole page first of all to find where they should go - it is not easy to find the exact sentence or section when dealing with translations because my departure point is the English text.

**Comment by mdlincoln** [comment]
@arojascastro You are right, the "changes" view is limited. This link https://github.com/programminghistorian/jekyll/blob/c1636fd17fa5c3f44bffa70a84011e1db1afa4c9/editor-guidelines.md should show the entire file.

You can navigate to that link for any Pull Request by clicking on the "View" button in the "Files Changed" tab:

screen shot 2017-05-27 at 9 26 08 am

**Comment by mdlincoln** [comment]
On hold until after redesign branch is merged, as there will need to be a few updates to the English text, and translation should happen on the new version of the Spanish editor guidelines page.

**Pull request merged**